addons/im_livechat/static/src/models/thread/thread.js

```diff
@@ -39,18 +39,17 @@ registerClassPatchModel('mail.thread', 'im_livechat/static/src/models/thread/thr
                  * of polluting the database, it is therefore acceptable and
                  * easier to handle one temporary partner per channel.
                  */
-                data2.members.push(unlink(this.env.messaging.publicPartners));
                 const partner = this.env.models['mail.partner'].create(
                     Object.assign(
                         this.env.models['mail.partner'].convertData(data.livechat_visitor),
                         { id: this.env.models['mail.partner'].getNextPublicId() }
                     )
                 );
-                data2.members.push(link(partner));
+                data2.members[0]._value[0].partner = link(partner);
```

Comment:
I have yet to check the base method, but this is unreadable. It would be better with some kind of named variables to understand it.

I think this wrong too, there is no guarantee that accessing `members[0]` will lead to selecting the member of the public partner.

Reply:
I don't know how to properly manipulate the command list that we get there, if you have an idea for that

Reply:
It's a bit crappy either way, but basically it will depend on the format returned in the super method.

In this case:
`data2.members` is a list of commands
`data2.members.length` can either be 0 (if no `members` is not given in `data`) or 1, so this needs to be verified
`data2.members[0]` is the first command, which is either `unlinkAll` or `insertAndReplace` (this needs to be verified, I guess `members` given in `data` is not supposed to be empty but there is no strong guarantee)
`data2.members[0]._value` could use the `value` getter instead of the private attribute. Also accessing value only makes sense if it is indeed `insertAndReplace` for parent. (`unlinkAll` or no command given will not have this)
`data2.members[0]._value[0]` is the first element to insert (based on the parent giving a list of data to the command). Instead of accessing the first element, it should be iterated on all of them to find those that have the public partner.